src/poetry/utils/exporter.py

```diff
@@ -120,11 +120,10 @@ def _export_requirements_txt(
                     hashes.append(f"{algorithm}:{h}")
 
                 if hashes:
-                    line += " \\\n"
+                    sep = " \\\n"
+                    line += sep
                     for i, h in enumerate(hashes):
-                        line += "    --hash={}{}".format(
-                            h, " \\\n" if i < len(hashes) - 1 else ""
-                        )
+                        line += f"    --hash={h}{sep if i < len(hashes) - 1 else ''}"
```

Comment:
I'm thinking this can be refactored to use `string.join()`... Something along the lines of `sep.join(f"--hash={h}" for h in hashes)`.